### PR TITLE
Add ExtInst validation pass (GLSL only for now)

### PR DESCRIPTION
**Pull request by atgoo**
Validates all GLSL.std.450 extended instructions.

### CODE REVIEW

File: source/validate_ext_inst.cpp [hunk 1]
```diff
@@ -0,0 +1,755 @@
...
+          }
+
+          if (result_type_bit_width != _.GetBitWidth(operand_type)) {
+            return _.diag(SPV_ERROR_INVALID_DATA)
+                   << ext_inst_name() << ": "
+                   << "expected all operands to have the same bit width as "
+                   << "Result Type";
+          }
+
+          if (ext_inst_key == GLSLstd450FindILsb ||
```
Comment (dneto0) [inline source/validate_ext_inst.cpp]: FindILsb does not have this restriction.
(Was confirmed by Khronos internal issue #173 if you want to look it up.)

Comment (atgoo) [reply source/validate_ext_inst.cpp]: It seems that the issue was left undecided.

File: source/validate_ext_inst.cpp [hunk 2]
```diff
@@ -0,0 +1,755 @@
...
+        uint32_t interpolant_storage_class = 0;
+        uint32_t interpolant_data_type = 0;
+        if (!_.GetPointerTypeInfo(interpolant_type, &interpolant_data_type,
+                                  &interpolant_storage_class)) {
+          return _.diag(SPV_ERROR_INVALID_DATA)
+                 << ext_inst_name() << ": "
+                 << "expected Interpolant to be a pointer";
+        }
+
+        if (result_type != interpolant_data_type) {
```
Comment (dneto0) [inline source/validate_ext_inst.cpp]: Yes, thanks for thinking this one through.  I've filed Khronos internal issue 260 to fix the description in the spec.

File: source/validate_ext_inst.cpp (hunk 1 above)
Comment (dneto0) [reply source/validate_ext_inst.cpp]: We decided to leave ILsb as it is currently specified, without the constraint.
The undecided part was whether to relax the UMsb SMsb instructions.   We're waiting for a request change it.

**Comment by dneto0** [comment]
Rebased, squashed, and pushed into master as 25d396b4a2

**Pull request closed without merging**